### PR TITLE
[chore] - update Docker source

### PR DESCRIPTION
**Pull request by ahrav**
### Description:
- Add common chunker to handle tar files.
- Add ability to scan images concurrently.

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/sources/docker/docker.go [hunk 1]
```diff
@@ -71,164 +72,202 @@ func (s *Source) Init(_ context.Context, name string, jobId, sourceId int64, ver
...
+					return err
+				}
+				dockerLayersScanned.WithLabelValues(s.name).Inc()
+			}
 
-		layers, err := img.Layers()
-		if err != nil {
-			return err
-		}
+			dockerImagesScanned.WithLabelValues(s.name).Inc()
```
Comment (rosecodym) [inline pkg/sources/docker/docker.go:121]: what's the additional increment for?

File: pkg/sources/docker/docker.go [hunk 2]
```diff
@@ -71,164 +72,202 @@ func (s *Source) Init(_ context.Context, name string, jobId, sourceId int64, ver
...
 
-		for _, layer := range layers {
+			return nil
+		})
+	}
+	_ = workers.Wait()
```
Comment (rosecodym) [inline pkg/sources/docker/docker.go:126]: is it generally considered ok to discard this result? (without even, say, logging it?)

File: pkg/sources/docker/docker.go [hunk 3]
```diff
@@ -71,164 +72,202 @@ func (s *Source) Init(_ context.Context, name string, jobId, sourceId int64, ver
...
 			}
-		}
 
-		logger = logger.WithValues("image", base, "tag", tag)
-		logger.V(2).Info("scanning image")
+			for _, layer := range layers {
+				if err := s.processLayer(ctx, layer, imgInfo, chunksChan); err != nil {
+					return err
+				}
+				dockerLayersScanned.WithLabelValues(s.name).Inc()
```
Comment (rosecodym) [inline pkg/sources/docker/docker.go:118]: you also increment this inside `processLayer` - was that on purpose?

**Review by rosecodym — COMMENTED** [review]
are there any tests you can add for the new tar functionality?

**Comment by ahrav** [comment]
> are there any tests you can add for the new tar functionality?

new functionality? I don't believe there should be any new functionality in terms of the reading. The new functionality should only be the concurrency. 🤔

File: pkg/sources/docker/docker.go (hunk 3 above)
Comment (ahrav) [reply pkg/sources/docker/docker.go:118]: whoops. Removed the increment in processLayer.

**Comment by rosecodym** [comment]
oh, am i misunderstanding
> Add common chunker to handle tar files.

?

File: pkg/sources/docker/docker.go (hunk 2 above)
Comment (ahrav) [reply pkg/sources/docker/docker.go:126]: Added some logic to collect errors from the goroutines.

File: pkg/sources/docker/docker.go (hunk 1 above)
Comment (ahrav) [reply pkg/sources/docker/docker.go:121]: I think this is the only increment for images. I can't find the other increment.

**Comment by ahrav** [comment]
> oh, am i misunderstanding
> 
> > Add common chunker to handle tar files.
> 
> ?

Oh, that shouldn't be new functionality. I am just using the common chunker to handle the chunking instead of doing it manually. The existing test should cover this behavior.

File: pkg/sources/docker/docker.go (hunk 1 above)
Comment (rosecodym) [reply pkg/sources/docker/docker.go:125]: oh whoops i missed that images has its own metric

**Review by rosecodym — APPROVED** [review]

**Pull request merged**